docs/strategies/rate-limiter.md

```diff
@@ -74,6 +74,140 @@ catch (RateLimiterRejectedException ex)
 | `DefaultRateLimiterOptions` | `PermitLimit` set to 1000 and `QueueLimit` set to 0. | The options for the default concurrency limiter that will be used when `RateLimiter` is `null`. |
 | `OnRejected`                | `null`                                               | Event that is raised when the execution is rejected by the rate limiter.                        |
 
+## Diagrams
+
+### Rate Limiter
+
+Let's suppose we have a rate limiter strategy with `PermitLimit` : `1` and `Window` : `10 seconds`.
+
+### Rate Limiter: happy path sequence diagram
+
+```mermaid
+%%{init: {'theme':'dark'}}%%
+sequenceDiagram
+    actor C as Caller
+    participant P as Pipeline
+    participant RL as RateLimiter
+    participant D as DecoratedUserCallback
+
+    C->>P: Calls ExecuteAsync
+    P->>RL: Calls ExecuteCore
+    Note over RL,D: Window start
+    RL->>+D: Invokes
+    D->>-RL: Returns result
+    RL->>P: Returns result
+    P->>C: Returns result
+    Note right of C: Several seconds later...
```

Comment:
I'd maybe put this comment _over_ the line of the caller, as that's who's controlling the rate. It currently looks like the delay is driven by the result being returned.